io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,356 @@
+    if (isRunning ())
+      return;
+
+    running_ = true;
+
+    rs2::config cfg;
+
+    // capture from file
+    if (!file_name_.empty ())
+    {
+      cfg.enable_device_from_file (file_name_);
+    }
+    else
+    {
+      if (serial_number_)
+        cfg.enable_device (std::to_string (serial_number_));
+
+      cfg.enable_stream (RS2_STREAM_COLOR, device_width_, device_height_, RS2_FORMAT_RGB8, target_fps_);
+      cfg.enable_stream (RS2_STREAM_DEPTH, device_width_, device_height_, RS2_FORMAT_ANY, target_fps_);
+      cfg.enable_stream (RS2_STREAM_INFRARED, device_width_, device_height_, RS2_FORMAT_ANY, target_fps_);
+    }    
+
+    pipe_.start (cfg);
+
+    thread_ = std::thread (&RealSense2Grabber::threadFunction, this);
+
+  }
+
+  void 
+  RealSense2Grabber::stop ()
+  {
+    std::lock_guard<std::mutex> guard (mutex_);
+
+    quit_ = true;
+    running_ = false;
+  }
+
+  bool 
+  RealSense2Grabber::isRunning () const
+  {
+    std::lock_guard<std::mutex> guard (mutex_);
+
+    return running_;
+  }
+
+  float 
+  RealSense2Grabber::getFramesPerSecond () const
+  {
+    return fps_;
+  }
+
+  pcl::PointCloud<pcl::PointXYZ>::Ptr 
+  RealSense2Grabber::convertDepthToPointXYZ (const rs2::points & points)
+  {
+    pcl::PointCloud<pcl::PointXYZ>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZ> ());
+
+    auto sp = points.get_profile ().as<rs2::video_stream_profile> ();
+    cloud->width = sp.width ();
+    cloud->height = sp.height ();
+    cloud->is_dense = false;
+    cloud->points.resize (points.size ());
+
+    auto vertices_ptr = points.get_vertices ();
+
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
+    {
+      auto ptr = vertices_ptr + index;
+      auto& p = cloud->points[index];
+
+      p.x = ptr->x;
+      p.y = ptr->y;
+      p.z = ptr->z;
+    }
+
+    return cloud;
+  }
+
+  pcl::PointCloud<pcl::PointXYZI>::Ptr 
+  RealSense2Grabber::convertInfraredDepthToPointXYZI (const rs2::points & points, rs2::video_frame & ir)
+  {
+    pcl::PointCloud<pcl::PointXYZI>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZI> ());
+
+    auto sp = points.get_profile ().as<rs2::video_stream_profile> ();
+    cloud->width = sp.width ();
+    cloud->height = sp.height ();
+    cloud->is_dense = false;
+    cloud->points.resize (points.size ());
+
+    auto vertices_ptr = points.get_vertices ();
+    auto texture_ptr = points.get_texture_coordinates ();
+
+    uint8_t r, g, b;
+
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
+    {
+      auto ptr = vertices_ptr + index;
+      auto uvptr = texture_ptr + index;
+      auto& p = cloud->points[index];
+
+      p.x = ptr->x;
+      p.y = ptr->y;
+      p.z = ptr->z;
+
+      std::tie (r, g, b) = getTextureColor (ir, uvptr->u, uvptr->v);
+
+      p.intensity = 0.299f * static_cast <float> (r) + 0.587f * static_cast <float> (g) + 0.114f * static_cast <float> (b);
```

Comment:
I think that this conversion process becomes unnecessary with specifying <code>RS2_FORMAT_Y8</code> format.

Reply:
I'd like to make this dynamic and handle UYVY and Y8 but I'm not sure how to read the stream format.  Also, if it is Y8, do you know how to extract the intensity data directly?

Reply:
In case of <code>Y8</code>, I think intensity will simply use pixel value.

Reply:
I updated so both Y8 and UYVY are handled.  Please let me know if anything looks off